common/persistence/cassandra/cassandraHistoryPersistence.go

```diff
@@ -93,6 +97,11 @@ func (h *cassandraHistoryPersistence) Close() {
 }
 
 func (h *cassandraHistoryPersistence) AppendHistoryEvents(request *p.InternalAppendHistoryEventsRequest) error {
+	if len(request.Events.Data) > h.transactionSizeLimit {
```

Comment:
I think we can enforce this at the common persistence layer, so we don't have to do it for each implementation of the store.